tests/func/test_import.py

```diff
@@ -725,12 +725,40 @@ def test_import_invalid_configs(tmp_dir, scm, dvc, erepo_dir):
         )
 
 
-def test_import_no_hash(tmp_dir, scm, dvc, erepo_dir, mocker):
+@pytest.mark.parametrize(
+    "files",
+    [
+        {"foo": "foo"},
+        {
+            "dir": {
+                "bar": "bar",
+                "subdir": {"lorem": "ipsum", "nested": {"lorem": "lorem"}},
+            }
+        },
+    ],
+)
+def test_import_no_hash(tmp_dir, scm, dvc, erepo_dir, mocker, files):
     with erepo_dir.chdir():
-        erepo_dir.dvc_gen("foo", "foo content", commit="create foo")
+        erepo_dir.dvc_gen(files, commit="create foo")
 
     spy = mocker.spy(hash, "file_md5")
-    stage = dvc.imp(os.fspath(erepo_dir), "foo", "foo_imported")
-    assert spy.call_count == 1
-    for call in spy.call_args_list:
-        assert stage.outs[0].fs_path != call.args[0]
+    index_info_spy = mocker.spy(DataIndex, "info")
+    name = next(iter(files))
+    dvc.imp(os.fspath(erepo_dir), name, "out")
+
+    local_hashes = [
+        call.args[0] for call in spy.call_args_list if call.args[1].protocol == "local"
+    ]
+    assert not local_hashes
```

Comment:
Fixed the assumption here to assert that we don't hash any files. I mistakenly assumed that we hash files once, but that's not correct. 

It passed because we hash the tree object internally in `MemoryFileSystem`.

Reply:
Yes, I think leaving `assert spy.call_count == 1` was either a mistake or I should have left a comment about it.

Reply:
Might want to rename `spy` to something like `hash_spy` to not confuse with `index_info_spy`.